src/action.py

```diff
@@ -276,15 +276,19 @@ def _run_index(
     if dry_run:
         report = types_.ActionReport(
             table_row=None,
-            url=DRY_RUN_NAVLINK_LINK if action.type_ == types_.ActionType.CREATE else action.url,
+            url=(
+                DRY_RUN_NAVLINK_LINK
+                if isinstance(action, types_.CreateIndexAction)
+                else action.url
+            ),
             result=types_.ActionResult.SKIP,
             reason=DRY_RUN_REASON,
         )
         logging.info("report: %s", report)
```

Comment:
Unrelated to PR, but may I know when to use "%s" and f-strings?

Reply:
https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#f-strings